### PR TITLE
Generate: Prevent ini Files from Being Included in YAML Discovery

### LINKED ISSUE #4124: Generate: ignore desktop.ini
### What feature would you like to see?

Windows may create a `desktop.ini` file in Players/ folder if it ever tries to render thumbnails of files in there, which then breaks generate. See [Discord](https://discord.com/channels/731205301247803413/1300970907581550654). We already ignore the mac equivalent `.DS_Store`  because it starts with a `.`.

[YAML discovery in Generate.main](https://github.com/ArchipelagoMW/Archipelago/blob/ad40acd392435b19962d6e2f99aab5848cccee4b/Generate.py#L111) should change to ignore either all *.ini or at least desktop.ini.

### PR DESCRIPTION
**Pull request by nicholassaylor**
## What is this fixing or adding?
Closes #4124 
Prevents ini files from being read into generation. This can be a problem if desktop.ini is created by Windows. This does not prevent any other file formats from being used.

## How was this tested?
Transplanting a desktop.ini file into the folder and running a generation with a valid yaml included

### CODE REVIEW

**Review by black-sliver — DISMISSED** [review]
Thanks!

**Review by black-sliver — COMMENTED** [review]
Oh, wait. I believe it'd be better if that check would be case-insensitive?

**Comment by Jouramie** [comment]
Out of curiosity, is there any reason to allow other file types than only yaml?

**Comment by palex00** [comment]
> Out of curiosity, is there any reason to allow other file types than only yaml?

.yaml, .yml and .txt all work for the sake of AP. And that's only the ones I am aware of.

**Comment by black-sliver** [comment]
json is also kinda valid

File: Generate.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def main(args=None) -> Tuple[argparse.Namespace, int]:
     player_files = {}
     for file in os.scandir(args.player_files_path):
         fname = file.name
-        if file.is_file() and not fname.startswith(".") and \
+        if file.is_file() and not fname.startswith(".") and not fname.endswith(".ini") and \
```
Comment (black-sliver) [inline Generate.py:113]:
```suggestion
        if file.is_file() and not fname.startswith(".") and not fname.lower().endswith(".ini") and \
```

**Review by black-sliver — APPROVED** [review]

**Pull request merged**